boa/src/object/mod.rs

```diff
@@ -603,6 +608,33 @@ impl Object {
         }
     }
 
+    #[inline]
+    pub fn expect_map_mut(&mut self) -> &mut OrderedMap<JsValue> {
+        self.as_map_mut().expect("object was not a map")
+    }
+
+    #[inline]
+    pub fn is_map_iterator(&self) -> bool {
+        self.as_map_iterator_ref().is_some()
```

Comment:
I think here we should use the `matches!()` macro, and that way we don't need to first extract the option and then check it's variant.